modules/dockermodelrunner/internal/sdk/client/pull.go

```diff
@@ -18,20 +20,37 @@ func (c *Client) PullModel(ctx context.Context, fullyQualifiedModelName string)
 		return fmt.Errorf("new post request (%s): %w", reqURL, err)
 	}
 
-	log.Default().Printf("🙏 Pulling model %s. Please be patient, no progress bar yet!", fullyQualifiedModelName)
+	logger := log.Default()
+
+	logger.Printf("🙏 Pulling model %s. Please be patient", fullyQualifiedModelName)
 
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		return fmt.Errorf("http post: %w", err)
 	}
 	defer resp.Body.Close()
 
+	// Check context after getting response
+	if err := ctx.Err(); err != nil {
+		return fmt.Errorf("context done after response: %w", err)
+	}
+
```

Comment:
suggestion: not sure if this was missed, but this isn't needed, the read from resp.Body by scanner will detect if needed.